### ISSUE TITLE
GDumb Training Loop

### ISSUE THREAD

**Issue by Confusezius** [aimagelab/mammoth#25]
First of all - thanks for the great codebase!

I do have a quick question. In the GDumb implementation , it seems the optimizer is placed inside of the training loop, which means that the scheduler does not update the optimizer learning rate, no?

https://github.com/aimagelab/mammoth/blob/cee3a43db4b4008ff6168fb308b49def928e58db/models/gdumb.py#L35

**Comment by mbosc** [comment]
Yes exactly, this is because GDumb trains only *after* each task, in an iid fashion. The current implementation features the same augmentation scheme and scheduling of their original paper. Any change to it must be implemented in its end_task function.

**Comment by Confusezius** [comment]
Appreciate the blazingly fast response!

I think my question was formulated a bit ambiguously - I was referring to the training happening in the `fit_buffer()` function, which has the structure

```
for epoch in epochs:
   optimizer = ...
   scheduler = ...

   scheduler.step()

  training stuff.
```

Should it look like this? Because that way, no learning rate scheduling is applied, as the scheduler is always re-initialized, no?
I.e. something like

```
optimizer = ...
scheduler = ...
for epoch in epochs:
   scheduler.step()
   training stuff
```

**Comment by mbosc** [comment]
Oh yeah, that should obviously be out of the loop. Thanks, I'll fix it asap
